02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,355 @@
+
+```{r}
+# We can use the `summary()` function to get a better idea of the distribution of symbols in the `Symbol` column
+summary(as.factor(human_symbol_df$human_symbol)) # We need to use the `as.factor()` function here in order to get the count of each unique value returned
+```
+
+Looks like we have mapped symbols!
+It does not explicitly tell us if we have IDs that did not map to symbols, which would result in NAs so let's explore to find out if we do have any.
+
+```{r}
+any(is.na(human_symbol_df$human_symbol)) # Logical statement to check if there are any NAs in the `human_symbol` column of our mapped data frame
+```
+
+There are no NAs!
+This means that all of our zebrafish Ensembl IDs mapped to at least one human gene symbol!
+This is exciting because it is not always the case everything maps across species and that is okay.
+
+However, if you have almost all NAs it is possible that the function was executed incorrectly or you may want to consider using a different gene identifier, if possible.
+
+## Write results to file
+
+Now let's write our results to file.
+
+```{r}
+readr::write_tsv(
+  human_symbol_df,
+  file.path(
+    results_dir,
+    "GSE71270_ensembl_to_human_gene_symbol.tsv" # Replace with a relevant output file name
+  )
+)
+```
+
+*Some things to note:*
+
+* Multiple zebrafish Ensembl gene IDs map to the same human symbol which results in duplicated human gene symbols.
+Some downstream tools may need this to be resolved.
+One way to resolve this may be to map to a different gene identifier.
+See our [gene ID annotation microarray notebook](https://alexslemonade.github.io/refinebio-examples/02-microarrary/gene-id-annotation_microarray_01_ensembl.html) for an example of mapping between gene identifiers.
+* If a zebrafish Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+Let's look at the `ENSDARG00000069142` example below.
+
+```{r}
+human_symbol_df %>%
+  dplyr::filter(zebrafish_ensembl == "ENSDARG00000069142")
+```
+
+We can collapse the multi-mapped values into a list for each Ensembl ID or we can keep it in the format above with one row for each mapped ID (resulting in duplicated rows for some Ensembl IDs).
+
+## Collapse mapped values
+
+In the next chunk, we show how we can collapse all the symbols into one column where we store them all for each unique Ensembl ID.
+
+```{r}
+collapsed_symbol_df <- human_symbol_df %>%
+  # Group by Ensembl IDs
+  dplyr::group_by(zebrafish_ensembl) %>%
+  # Collapse the mapped values in `human_symbol_df` into one column named `all_human_symbols`
+  dplyr::summarize(all_human_symbols = paste(human_symbol, collapse = ";"))
+
+head(collapsed_symbol_df)
+```
+
+Since multiple zebrafish Ensembl gene IDs map to the same human symbol, we can also collapse all Ensembl IDs that each unique gene symbol mapped to.
+
+```{r}
+collapsed_ensembl_df <- human_symbol_df %>%
```

Comment:
Currently, this doesn't end up having the data associated with it, and I'm hesitant to combine values across zebrafish transcripts using an average, because that is a lot of data to lose and will likely not be representative of the zebrafish biology (which is an inherent issue with ortholog mapping).